### ISSUE TITLE
Error when using same location with variables for devices without generator and APNonce

### ISSUE THREAD

**Issue by mattp12** [airsquared/blobsaver#519]
**Describe the bug**
In version 3.4.0 you now can use certain variables in the path to better organize the blobs you save.
However, when using the same path (with variables) for multiple devices, an error occurs.

For instance, the path
`C:\Users\Matt\Blobs\${DeviceModel}\${MajorVersion}\${FullVersionString}`
works for saving my iPhone 12 blobs just fine and the folders are created.

When I set this to be the path of my iPhone 8 Plus as well, clicking Go does nothing but give an error shown in the debug log. 

I would imagine this is supposed to work, as the variables would create new folders based on the device (a new folder called "iPhone 12 (iPhone13,2)" was created, for instance), rather than somehow conflict with each other.

The log is below.

**Screenshots**
N/A

**Desktop:**
 - OS Version: Windows 11
 - iOS Device: N/A

**Checklist (place an x between the brackets to mark as completed):**
 - [x] I have made sure I am on the latest version of blobsaver
 - [x] I have checked other open issues to see if my problem has already been reported
 - [x] I have tried common troubleshooting tactics such as restarting my computer and reinstalling blobsaver

**Log(s)**
```
Exception in thread "JavaFX Application Thread" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at javafx.fxml@19/javafx.fxml.FXMLLoader$MethodHandler.invoke(Unknown Source)
	at javafx.fxml@19/javafx.fxml.FXMLLoader$ControllerMethodEventHandler.handle(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at javafx.base@19/javafx.event.Event.fireEvent(Unknown Source)
	at javafx.graphics@19/javafx.scene.Node.fireEvent(Unknown Source)
	at javafx.controls@19/javafx.scene.control.Button.fire(Unknown Source)
	at javafx.controls@19/com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(Unknown Source)
	at javafx.controls@19/com.sun.javafx.scene.control.inputmap.InputMap.handle(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at javafx.base@19/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at javafx.base@19/javafx.event.Event.fireEvent(Unknown Source)
	at javafx.graphics@19/javafx.scene.Scene$MouseHandler.process(Unknown Source)
	at javafx.graphics@19/javafx.scene.Scene.processMouseEvent(Unknown Source)
	at javafx.graphics@19/javafx.scene.Scene$ScenePeerListener.mouseEvent(Unknown Source)
	at javafx.graphics@19/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at javafx.graphics@19/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at javafx.graphics@19/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(Unknown Source)
	at javafx.graphics@19/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(Unknown Source)
	at javafx.graphics@19/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(Unknown Source)
	at javafx.graphics@19/com.sun.glass.ui.View.handleMouseEvent(Unknown Source)
	at javafx.graphics@19/com.sun.glass.ui.View.notifyMouse(Unknown Source)
	at javafx.graphics@19/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at javafx.graphics@19/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.sun.javafx.reflect.Trampoline.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at javafx.base@19/com.sun.javafx.reflect.MethodUtil.invoke(Unknown Source)
	at javafx.fxml@19/com.sun.javafx.fxml.MethodHelper.invoke(Unknown Source)
	... 47 more
Caused by: java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Unknown Source)
	at java.base/java.util.ImmutableCollections$MapN.<init>(Unknown Source)
	at java.base/java.util.Map.of(Unknown Source)
	at airsquared.blobsaver@3.4.0/airsquared.blobsaver.app.TSS.parsePath(Unknown Source)
	at airsquared.blobsaver@3.4.0/airsquared.blobsaver.app.TSS.<init>(Unknown Source)
	at airsquared.blobsaver@3.4.0/airsquared.blobsaver.app.TSS$Builder.build(Unknown Source)
	at airsquared.blobsaver@3.4.0/airsquared.blobsaver.app.Controller.createTSS(Unknown Source)
	at airsquared.blobsaver@3.4.0/airsquared.blobsaver.app.Controller.goButtonHandler(Unknown Source)
	... 54 more
```

**Comment by airsquared** [comment]
This error happens when you don't specify an APNonce and Generator. As a temporary fix until an updated version is released, you can specify an APNonce and Generator to fix the issue.

**Comment by 420Dope** [comment]
I have maybe the same issue on Windows Server.
The folder was not created when I use variables. The debug log is exactly the same as above (compared it with notepad++).

**Comment by airsquared** [comment]
This has been fixed in v3.4.1.
